### PR TITLE
Local Dev Web Server Enhancements

### PR DESCRIPTION
**Pull request by aj-stein-nist**
# Committer Notes

Some container improvements for local web development and testing for PRs. Closes usnistgov/OSCAL#1102.

### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/usnistgov/OSCAL/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/usnistgov/OSCAL/pulls) for the same update/change?
- [x] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [x] Do all automated CI/CD checks pass?

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- ~Have you written new tests for your core changes, as applicable?~
- ~Have you included examples of how to use your new feature(s)?~
- ~Have you updated all [OSCAL website](https://pages.nist.gov/OSCAL) and readme documentation affected by the changes you made? Changes to the OSCAL website can be made in the docs/content directory of your branch.~

### CODE REVIEW

File: docs/run-server.sh [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
+"${OSCAL_WORKING_PATH}/build/ci-cd/generate-specification-documentation.sh" -w "${OSCAL_WORKING_PATH}" 
+
+retval=$?
+
+if [ $retval -gt 0 ]; then
+  echo "Generating spec docs failed with error ${retval}, not running test site, review logs"
+  exit 1
+fi
+
+(cd "${OSCAL_WORKING_PATH}/docs" && hugo server $HUGO_ARGS)
```
Comment (bradh) [inline docs/run-server.sh:15]: New line at end of file:
```suggestion
(cd "${OSCAL_WORKING_PATH}/docs" && hugo server $HUGO_ARGS)

```

**Comment by aj-stein-nist** [comment]
It's [always fun when tests fail because of downloading some JAR or package](https://github.com/usnistgov/OSCAL/runs/4922650059?check_suite_focus=true) but clicking the URL in the logs works for me. Rerunning!

**Comment by aj-stein-nist** [comment]
Talked with Dave for a brief speed review this morning. I am not sure why `hugo` was not effectively watching before, it [_does_ do that by default](https://gohugo.io/commands/hugo_server/#options). Regardless, argeed we ought to mark it draft until I make some more of the changes I described. Will look into why `--watch` is flakey in my local env.

**Review by david-waltermire-nist — APPROVED** [review]
LGTM thanks!

**Pull request merged**